### PR TITLE
fix(FR-2355): handle duplicate TOML keys in config.toml parsing

### LINKED ISSUE #6091: Login config.toml options not applied after Lit-to-React migration
During the Lit to React migration, several config.toml login-related configuration options are not being properly applied on the frontend. The following options need to be audited and implemented in the React codebase:

- allowChangeSigninMode
- signupSupport
- allowSignout
- allowAnonymousChangePassword
- allowSignupWithoutConfirmation
- loginAttemptLimit
- loginBlockTime

Steps:

1. Identify which of the above options were handled in the old Lit shell
1. Find the corresponding React components (LoginFormPanel, LoginView)
1. Verify each option is read from config and applied correctly
1. Implement any missing handling

JIRA Issue: FR-2355

### PR DESCRIPTION
**Pull request by ironAiken2**
Resolves #6091 ([FR-2355](https://lablup.atlassian.net/browse/FR-2355))

## Summary
- `smol-toml` throws on duplicate keys in TOML, but the previous code silently swallowed the error, causing all config options to fall back to defaults without any indication to the user
- Change `fetchAndParseConfig` return type from `RawTomlConfig | null` to `{ config, error }` to distinguish fetch failures (network error, non-200, SPA fallback) from parse failures (invalid TOML)
- Add `configParseErrorState` Jotai atom to store parse errors so other hooks can access them
- Surface parse errors on the diagnostics page as a **critical** alert with the raw error message displayed in a `<pre>` code block, remediation instructions, and early return (skip remaining config checks since `rawConfig` is `null`)
- Log parse errors via `useBAILogger` (replaces `console.warn`)
- Update call sites (`loginSessionAuth.ts`, `EduAppLauncher.tsx`) for the new return type
- Update all tests for the new return type and add test cases for duplicate keys and parse failure
- Add i18n keys (`ConfigParseError`, `ConfigParseErrorDesc`, `ConfigParseErrorFix`) across all 21 languages

## Test plan
- [ ] Add duplicate keys in `config.toml` (e.g., `debug = false` and `debug = true` under `[general]`), open the diagnostics page, and verify a **critical** alert is shown with the parse error message in a code block
- [ ] Verify config values are applied correctly when `config.toml` is valid
- [ ] Verify the app falls back to defaults without errors when `config.toml` is missing (404)
- [ ] `pnpm run test` — all 35 useWebUIConfig tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[FR-2355]: https://lablup.atlassian.net/browse/FR-2355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: react/src/hooks/useWebUIConfig.ts [hunk 1]
```diff
@@ -113,32 +114,100 @@ function preprocessToml(config: RawTomlConfig): void {
...
+  }
+
+  if (duplicateIndices.size === 0) return text;
+
+  // Second pass: build result without duplicates
+  for (let i = 0; i < lines.length; i++) {
+    if (!duplicateIndices.has(i)) {
+      result.push(lines[i]);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply react/src/hooks/useWebUIConfig.ts]: Good point. Fixed the reverse-traversal bug by rewriting with a forward-pass approach that correctly tracks section boundaries. The function now walks forward to assign sections per-line and record the last key index per (section, key), then builds the result skipping earlier duplicates.

File: react/src/hooks/useWebUIConfig.ts [hunk 2]
```diff
@@ -113,32 +114,100 @@ function preprocessToml(config: RawTomlConfig): void {
...
+    const sanitizedText = deduplicateTomlKeys(text);
+    const parsed = toml(sanitizedText) as RawTomlConfig;
     preprocessToml(parsed);
-    return parsed;
-  } catch {
-    return null;
+    return { config: parsed };
+  } catch (error) {
+    return { config: null, error };
   }
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply react/src/hooks/useWebUIConfig.ts:168]: Good point. Separated fetch vs parse error handling — network/fetch errors now return `{ config: null }` (no error field), while only TOML parse failures set the error field. The log message "Failed to parse" is now accurate since it only triggers for actual parse errors.

File: react/src/hooks/useWebUIConfig.test.ts [hunk 3]
```diff
@@ -170,19 +172,33 @@ page = ""
...
+debug = false
+apiEndpoint = "https://api.example.com"
+debug = true
+`;
+    mockFetch(200, tomlContent);
+    const result = await fetchAndParseConfig('/config.toml');
+    expect(result.config).not.toBeNull();
+    expect(result.config?.general?.debug).toBe(true);
+    expect(result.error).toBeUndefined();
   });
```
[the remark this answers is not in the document]
Comment (ironAiken2) [reply react/src/hooks/useWebUIConfig.test.ts]: Good point. Added a multi-section regression test that covers duplicates in root-level keys, `[general]`, and `[wsproxy]` sections simultaneously, verifying that deduplication is correctly scoped per-section.

File: react/src/hooks/useWebUIConfig.ts [hunk 4]
```diff
@@ -113,32 +114,135 @@ function preprocessToml(config: RawTomlConfig): void {
...
+}
+
+/**
+ * Remove duplicate keys within the same TOML table/section.
+ * When the Backend.AI webserver generates config.toml from webserver.conf,
+ * it may produce duplicate keys (e.g. two `debug = true` lines) which
+ * violate the TOML spec and cause strict parsers like smol-toml to throw.
+ * This function keeps the last occurrence of each duplicate key.
+ */
+function deduplicateTomlKeys(text: string): string {
```
Comment (yomybaby) [inline react/src/hooks/useWebUIConfig.ts]: ou're implementing partial TOML parsing logic to handle duplicates, which means parsing twice. This doesn't seem ideal.

Questions:

Can the webserver actually send TOML with duplicate keys?
If duplicate keys shouldn't occur, for development, wouldn't it be better to let the parser error out (alerting us to the problem) rather than silently deduplicating?

**Review by yomybaby — DISMISSED** [review]

File: react/src/hooks/useWebUIConfig.ts (hunk 4 above)
Comment (ironAiken2) [reply react/src/hooks/useWebUIConfig.ts]: In fact, there was an issue where the web server was serving some TOML entries redundantly, which meant that even if we changed the settings in `webserver.toml`, the parser couldn't parse them correctly, so the changes weren't applied. (https://github.com/lablup/backend.ai/pull/10423)  
  
As you mentioned, rather than performing the parsing twice, I think it would be better to modify the code so that it displays an error only if the parser fails to parse the data correctly. I will make the necessary changes.
